### DISCUSSION TITLE
v5.1

### DISCUSSION THREAD

**Discussion by snakers4** [snakers4/silero-vad#494] [General]
# Experimental PIP package release

- Experimental pip-package release;
- Community PRs to update the examples;
 
## What's Changed

* Adamnsandle by @adamnsandle in https://github.com/snakers4/silero-vad/pull/481
* Update microphone_and_webRTC_integration.py by @eltociear in https://github.com/snakers4/silero-vad/pull/475
* cpp example by @filtercodes in https://github.com/snakers4/silero-vad/pull/482
* Update Golang example to support model v5 by @streamer45 in https://github.com/snakers4/silero-vad/pull/489
* Create python-publish.yml by @adamnsandle in https://github.com/snakers4/silero-vad/pull/492
* Adamnsandle by @adamnsandle in https://github.com/snakers4/silero-vad/pull/493

## New Contributors
* @eltociear made their first contribution in https://github.com/snakers4/silero-vad/pull/475
* @filtercodes made their first contribution in https://github.com/snakers4/silero-vad/pull/482

**Full Changelog**: https://github.com/snakers4/silero-vad/compare/v5.0...v5.1

<hr /><em>This discussion was created from the release <a href='https://github.com/snakers4/silero-vad/releases/tag/v5.1'>v5.1</a>.</em>

**Comment by lifeiteng** [comment]
@snakers4 Thanks for your work!

How do I knowing if I'm using the latest version?

```
model, utils = torch.hub.load(repo_or_dir='snakers4/silero-vad', model='silero_vad')
```

**Reply by snakers4** [reply]
Torch hub load without tags or branches just load the latest state of the repo

**Comment by ozancaglayan** [comment]
Thanks!

Just as a side info: I now tried this version on a sample of private audio files and compared `v4` and `v5`. It seems that `v5` rejected around 25% more content given the same settings. Upon listening to some of those rejected regions, I hear proper spoken content (sometimes unintelligible sometimes intelligible but children voices)  It would be good if you can give us some high-level description of what kind of data the model was trained on and how did it change from `v4`

**Reply by snakers4** [reply]
It is always worth doing the following:

- Isolating the problematic audio;
- Plotting the probability chart (there is param in the `get_speech_timestamps` function);
- Seeing if the problem is systemic, or just improper hyper-paramers;

> It would be good if you can give us some high-level description of what kind of data the model was trained on and how did it change from v4

We added as diverse data as possible for 6k languages, and put major focus on productions data, noises, edge cases.
We also tested the VAD on many more validation sets.

> I hear proper spoken content (sometimes unintelligible sometimes intelligible but children voices)

Can you share some audios?

**Reply by ozancaglayan** [reply]
Yes, I'll do the analytics at some point, thanks!

Unfortunately, I can't share these audio files but I'm pretty sure I/we can find some examples with children speaking.

**Reply by leminhnguyen** [reply]
@snakers4 I have the same problem, how can I rollback to V4 version?

**Comment by ozancaglayan** [comment]
Hi. I can't share some audios but can share plots of two audio files, under same settings with v4 and v5.1. The plot is the probability difference e.g. `v5_probs - v4_probs`. Positive values show the regions where v5 assigns more voice probability than v4 and vice versa. Both models are run with `window_size_samples=512`, and under ONNX.

<img width="1080" alt="image" src="https://github.com/user-attachments/assets/77e40d52-8049-4df5-82ed-32ba00da2e97">
<img width="1071" alt="image" src="https://github.com/user-attachments/assets/12d9526e-c409-4268-92ea-3e4691c51d01">

First one is especially interesting. But both plots show that the issue is not a simple threshold shift happening across the models. I wonder whether this is all because of the removed Adaptive Audio Normalization layer?

**Comment by TYMA22389** [comment]
Субтитры сделал DimaTorzok
